build/npm/preinstall.js

```diff
@@ -63,13 +63,24 @@ function hasSupportedVisualStudioVersion() {
 			break;
 		}
 		const programFiles86Path = process.env['ProgramFiles(x86)'];
+		const programFiles64Path = process.env['ProgramFiles'];
+
 		if (programFiles86Path) {
 			vsPath = `${programFiles86Path}/Microsoft Visual Studio/${version}`;
 			const vsTypes = ['Enterprise', 'Professional', 'Community', 'Preview', 'BuildTools'];
 			if (vsTypes.some(vsType => fs.existsSync(path.join(vsPath, vsType)))) {
 				availableVersions.push(version);
 				break;
 			}
+
+			else if (programFiles64Path) {
```

Comment:
Can this be an additional if case outside of the `if (programFiles86Path)` block, rather than an else-if here? Also, the body can be refactored into a separate function, considering how they're doing the same thing, just that one case uses `programFiles86Path` and the other uses `programFiles64Path`.

Reply:
We could check for programFiles64Path before we check for the 86 one because it should use VS2022 by default.

Reply:
> Can this be an additional if case outside of the `if (programFiles86Path)` block, rather than an else-if here? Also, the body can be refactored into a separate function, considering how they're doing the same thing, just that one case uses `programFiles86Path` and the other uses `programFiles64Path`.

Hmmm..  I think this will be more better.. working on it.

Reply:
### I tested the code and it worked fine.

```js 

		if (programFiles64Path) {
			vsPath = `${programFiles64Path}/Microsoft Visual Studio/${version}`;
			const vsTypes = ['Enterprise', 'Professional', 'Community', 'Preview', 'BuildTools'];
			if (vsTypes.some(vsType => fs.existsSync(path.join(vsPath, vsType)))) {
				availableVersions.push(version);
				break;
			}
		}

		if (programFiles86Path) {
			vsPath = `${programFiles86Path}/Microsoft Visual Studio/${version}`;
			const vsTypes = ['Enterprise', 'Professional', 'Community', 'Preview', 'BuildTools'];
			if (vsTypes.some(vsType => fs.existsSync(path.join(vsPath, vsType)))) {
				availableVersions.push(version);
				break;
			}
		}

```

Reply:
### inside a functions 
```js 

		function findWhereIsVisualStudio() {
			if (programFiles64Path) {
				vsPath = `${programFiles64Path}/Microsoft Visual Studio/${version}`;
				const vsTypes = ['Enterprise', 'Professional', 'Community', 'Preview', 'BuildTools'];
				if (vsTypes.some(vsType => fs.existsSync(path.join(vsPath, vsType)))) {
					availableVersions.push(version);
				}
			}

			if (programFiles86Path) {
				vsPath = `${programFiles86Path}/Microsoft Visual Studio/${version}`;
				const vsTypes = ['Enterprise', 'Professional', 'Community', 'Preview', 'BuildTools'];
				if (vsTypes.some(vsType => fs.existsSync(path.join(vsPath, vsType)))) {
					availableVersions.push(version);
				}
			}
		}
```

Reply:
@rzhao271 shall I update the code in this way

Reply:
removed the break statement

Reply:
I prefer the one where you don't move it out into a function.
We want to keep the break statements.